### PR TITLE
Add ReversibleStackPanel widget

### PR DESCRIPTION
**Pull request by ConradRoesch**
Rather than adding new reversable stack controls, I've moved the VStack/HStack constructors to from StackPanel to ReversibleStackPanel, so to accommodate the optional parameter 'reverseOrder'. Is this an appropriate design decision?

Also added StackPanel samples to the Gallery which demonstrate Spacing and ReverseOrder for vertical and horizontal panels

*Bug Found*
ReverseOrder applies correctly in the initial model, but the rendered view does not update if this property is subsequently changed (while the Spacing property does behave as expected). I believe that the bug is not in the ReversibleStackPanel wrapper, but rather the diff functions that determines what portions of the view need to be re-rendered, though the inner workings of this are a bit out of my depth at the moment.

### CODE REVIEW

File: src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    let WidgetKey = Widgets.register<ReversibleStackPanel> ()
+
+    let ReverseOrder =
+        Attributes.defineAvaloniaPropertyWithEquality ReversibleStackPanel.ReverseOrderProperty
+
+[<AutoOpen>]
+module ReversibleStackPanelBuilders =
+    type Fabulous.Avalonia.View with
+
+        static member VStack<'msg>(?spacing: float, ?reverseOrder: bool) =
```
Comment (edgarfgp) [inline src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs:22]:
```suggestion
        static member inline VStack<'msg>(?spacing: float, ?reverseOrder: bool) =
```

File: src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                     StackPanel.Spacing.WithValue(spacing),
+                     ReversibleStackPanel.ReverseOrder.WithValue(reverseOrder))
+
+            CollectionBuilder<'msg, IFabReversibleStackPanel, IFabControl>(
+                ReversibleStackPanel.WidgetKey,
+                scalers,
+                Panel.Children
+            )
+
+        static member HStack<'msg>(?spacing: float, ?reverseOrder: bool) =
```
Comment (edgarfgp) [inline src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs:46]:
```suggestion
        static member inline HStack<'msg>(?spacing: float, ?reverseOrder: bool) =
```

File: src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                match spacing with
+                | Some spacing -> spacing
+                | None -> 0.
+
+            let reverseOrder =
+                match reverseOrder with
+                | Some reverseOrder -> reverseOrder
+                | None -> false
+
+            let scalers: StackAllocatedCollections.StackList.StackList<ScalarAttribute> =
```
Comment (edgarfgp) [inline src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs:57]:
```suggestion
            let scalars =
                StackList.StackList.three
                    (StackPanel.Orientation.WithValue(Orientation.Vertical),
                     StackPanel.Spacing.WithValue(spacing),
                     ReversibleStackPanel.ReverseOrder.WithValue(reverseOrder))

            CollectionBuilder<'msg, IFabReversibleStackPanel, IFabControl>(
                ReversibleStackPanel.WidgetKey,
                scalars,
                Panel.Children
            )
```

File: src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                | Some spacing -> spacing
+                | None -> 0.
+
+            let reverseOrder =
+                match reverseOrder with
+                | Some reverseOrder -> reverseOrder
+                | None -> false
+
+
+            let scalers: StackAllocatedCollections.StackList.StackList<ScalarAttribute> =
```
Comment (edgarfgp) [inline src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs:34]: See the comment bellow :)

**Review by edgarfgp — COMMENTED** [review]
Yeah it makes sense to have all in one place . Just added couple comments. Great job 👍

File: src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+[<AutoOpen>]
+module ReversibleStackPanelBuilders =
+    type Fabulous.Avalonia.View with
+
+        static member VStack<'msg>(?spacing: float, ?reverseOrder: bool) =
+            let spacing =
+                match spacing with
+                | Some spacing -> spacing
+                | None -> 0.
```
Comment (TimLariviere) [inline src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs:26]: Optimization: When there is a `match ... with` condition on an optional argument in an inlined method, the F# compiler will resolve the condition at compile team and only keep the resulting value.

Say you write `VStack(spacing = 12.) { }`, the conditions will be removed at compile time and the code will look like:
```fs
let spacing = 12.
let reverseOrder = false

let scalers = (...)
```

We can take advantage of this optimization to improve even more the performance by avoiding to add the attribute to the widget altogether if the optional arguments are not passed.

This is possible by rewriting the code slightly differently:

```fs
static member inline VStack<'msg>(?spacing: float, ?reverseOrder: bool) =

    let mutable scalers = StackList.one(StackPanel.Orientation.WithValue(Orientation.Vertical))

    match spacing with
    | None -> ()
    | Some v -> scalers <- StackList.add(&scalers, StackPanel.Spacing.WithValue(spacing))

    match reverseOrder with
    | None -> ()
    | Some v -> scalers <- StackList.add(&scalers, ReversibleStackPanel.ReverseOrder.WithValue(reverseOrder))

    CollectionBuilder<'msg, IFabReversibleStackPanel, IFabControl>(
        ReversibleStackPanel.WidgetKey,
        scalers,
        Panel.Children
    )
```

So if you pass no argument, the code will be
```fs
    let scalers = StackList.one(StackPanel.Orientation.WithValue(Orientation.Vertical))

    CollectionBuilder<'msg, IFabReversibleStackPanel, IFabControl>(
        ReversibleStackPanel.WidgetKey,
        scalers,
        Panel.Children
    )
```

which is less work for Fabulous

Comment (ConradRoesch) [reply src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs]: Thank you for taking the time to explaining that in detail! I've just committed the changes you recommended.

File: src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs (hunk 2 above)
Comment (ConradRoesch) [reply src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs]: Thanks, I've committed these changes

**Review by edgarfgp — APPROVED** [review]

**Pull request merged**